### PR TITLE
Solutions for Chapter 4

### LINKED ISSUE #2: Add solution for `ch04`
Make `npm run ch04` run in CI and disallow merge when fails.

### PR DESCRIPTION
**Pull request by EmCeeEs**
Fix #2.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ jobs:
       - run:
           command: npm run test
           <<: *exercise_dir
+      - run:
```
Comment (EmCeeEs) [inline .circleci/config.yml]: Add chapter specific tests here for clarity.

**Comment by Bakkhos** [comment]
if this were a real pull request on a software project and not a bunch of personal exercises, you should...

**Comment by Bakkhos** [comment]
1.) Remove the WIP from the name of the pull request after it's done

**Comment by Bakkhos** [comment]
If I understand our convention correctly, WIP indicates that the reviewer shouldn't be looking at it, since it's not ready yet

**Comment by Bakkhos** [comment]
2.) Pull request names should be close to full english sentences. In this case maybe "Add solutions to exercises in Ch04"

**Comment by Bakkhos** [comment]
Branches that you work on, and that you intend to be merged into master, should normally be called feature/XYZ. I understand you don't want to merge the solutions, though.

**Comment by Bakkhos** [comment]
ok now to actually read the contents ... :)

File: exercises/ch04/exercise_a.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
 // Refactor to remove all arguments by partially applying the function.
 
 // words :: String -> [String]
-const words = str => split(' ', str);
+const words = split(' ');
```
Comment (Bakkhos) [inline exercises/ch04/exercise_a.js:4]: Looks good

File: exercises/ch04/exercise_b.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
 // Refactor to remove all arguments by partially applying the functions.
 
 // filterQs :: [String] -> [String]
-const filterQs = xs => filter(x => x.match(/q/i), xs);
+const filterQs = filter(match(/q/i));
```
Comment (Bakkhos) [inline exercises/ch04/exercise_b.js:4]: Looks good too

**Pull request closed without merging**